### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by jeromekelleher**
1.  Updated example tree sequence figure.
2. Implemented wf algorithm in sims/wf-figures.py and fixed some bugs in listing.
3.  Draft before and after simplification figure showing trees.
4. Rough draft finishing out the section with a discussion of mutations.
5. Very rough draft of some text explaining simplify. 

There's nothing very polish here I'm afraid, just initial passes at various parts of the document. There's more comments inline for where there are specific problems.

### CODE REVIEW

File: forwards_paper.tex [hunk 1]
```diff
@@ -186,8 +187,7 @@ \subsection*{Tree sequences}
         edges that define these topologies. The right-hand panels show the specific encoding
         of this tree sequence in the four tables (nodes, edges, sites and mutations) defined
         by \msprime.
-        \jk{The figure needs to be rescaled a bit to give more space to the trees. We could also use a
-        time scale next to the trees.}
+        \jk{Some minor alignment issues present here.}
```
Comment (petrelharp) [inline forwards_paper.tex:190]: looks pretty good to me.  I can always do final tweaks in inkscape if we want.

File: forwards_paper.tex [hunk 2]
```diff
@@ -335,17 +330,97 @@ \subsection*{Recording the pedigree in forwards time}
 from $x$ to $L$ is $P_b$. We then complete the replacement event by incrementing
 $n$, ready to represent the next new node.
 
+This algorithm records only the topological information resulting from the
+forwards in time Wright-Fisher process, but it is straightforward to add
+mutational information. This can be done in two different ways.
```
Comment (petrelharp) [inline forwards_paper.tex:335]: should clarify that if mutations are selected then we'd need to do it on the fly.

File: forwards_paper.tex [hunk 3]
```diff
@@ -335,17 +330,97 @@ \subsection*{Recording the pedigree in forwards time}
...
+    \begin{center}
+        \includegraphics{wf-before}
+        \includegraphics{wf-after}
+    \end{center}
+    \caption{An example of a marginal genealogy from a Wright-Fisher simulation
+    with $N=5$. On the left we have the original tree including all
+    intermediate nodes and dead-ends. On the right we have the minimal tree
+    relating all of the currently-alive individuals (27--31).
+    \jk{This figure needs a little tidying up. We should combine the two SVGs
+        into a single image with some (a) (b) annotations. Also the node labels
```
Comment (petrelharp) [inline forwards_paper.tex:361]: I can easily tweak these things in inkscape, but will wait 'til the last minute to do it.

File: forwards_paper.tex [hunk 4]
```diff
@@ -373,52 +447,106 @@ \subsection*{Recording the pedigree in forwards time}
...
+set of individuals, where each ancestral individual is composed of a
+linked list of these segments. The node $u$ in each ancestral sequence
+is the node ID associated with this segment in the output tree sequence.
+We maintain a map $A$ such that $A_j$ is segment chain for node $j$ in
+the input tree sequence.
+
+\jk{I'm not sure this is the right level of detail to tackle this description
+at. What's the goal of this algorithm description? Are we trying to
+establish some properties of the algorithm or just give the reader a high-level
+flavour of how it works??}
```
Comment (petrelharp) [inline forwards_paper.tex:483]: At least a high-level flavour - this is a good start.  I think it'd be nice to outline the steps in at least a bit more detail, to make it more concrete.  I suppose it also depends where we submit this, how much detail is optimal.

File: forwards_paper.tex (hunk 1 above)
Comment (jeromekelleher) [reply forwards_paper.tex:190]: I'll twiddle the alignments of a few columns at some point, but I think it's basically OK for now.

File: forwards_paper.tex (hunk 3 above)
Comment (jeromekelleher) [reply forwards_paper.tex:361]: That would be great, thanks Peter!

File: forwards_paper.tex (hunk 4 above)
Comment (jeromekelleher) [reply forwards_paper.tex:483]: This is a good question, which we should probably resolve soonish. I don't have any strong views, I'm almost certainly happy with whatever you think is best.

File: forwards_paper.tex (hunk 2 above)
Comment (jeromekelleher) [reply forwards_paper.tex:335]: That's true. I'm not entirely happy with the flow of thought through this section. It'd probably be good to talk it through and see if we can come up with a clear narrative.

**Pull request merged**